src/main/java/com/wolox/training/config/AuthConfig.java

```diff
@@ -0,0 +1,52 @@
+package com.wolox.training.config;
+
+import com.wolox.training.providers.UserAuthProvider;
+import com.wolox.training.services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class AuthConfig extends WebSecurityConfigurerAdapter {
+
+  @Autowired
+  public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
```

Comment:
This is not the correct method to configure AuthenticationManagerBuilder, change the signature of the method and remove the @Autorwired. Must be: `public void configure(AuthenticationManagerBuilder auth)`

The reason is that when you don't override `configure(AuthenticationManagerBuilder)` and have an `AuthenticationProvider` bean, it will be registered by Spring Security, you don't have to do anything else.

However, when `configure(AuthenticationManagerBuilder)` overridden, Spring Security will invoke it and won't try to register any provider by itself.
If you're curious, you can take a look the [related method][1]. `disableLocalConfigureAuthenticationBldr` is true if you override `configure(AuthenticationManagerBuilder)`.

So, briefly, if you want to register just one custom `AuthenticationProvider` then do not override `configure(AuthenticationManagerBuilder)`, do not call `authenticationProvider(AuthenticationProvider)` in `configure(HttpSecurity)`, just make your `AuthenticationProviver` implementation bean by annotating `@Component` and you're good to go. 


  [1]: https://github.com/spring-projects/spring-security/blob/master/config/src/main/java/org/springframework/security/config/annotation/web/configuration/WebSecurityConfigurerAdapter.java#L263